sql/auth/sql_user_table.cc

```diff
@@ -1213,6 +1213,10 @@ int replace_column_table(THD *thd, GRANT_TABLE *g_t, TABLE *table,
       store_record(table, record[1]);  // copy original row
     }
 
+    timeval tm;
+    tm = thd->query_start_timeval_trunc(0);
+    table->field[5]->store_timestamp(&tm);
```

Comment:
can we merge definition and assignment?
timeval tm = thd->query_start_timeval_trunc(0)

Reply:
I tried doing that. But it fails with 
```
[ 79%] Building CXX object sql/CMakeFiles/sql_main.dir/auth/sql_user_table.cc.o
/home/venki/work/ps/7617/sql/auth/sql_user_table.cc: In function 'int replace_routine_table(THD*, GRANT_NAME*, TABLE*, const LEX_USER&, const char*, const char*, bool, ulong, bool)':
/home/venki/work/ps/7617/sql/auth/sql_user_table.cc:1693:1: error: jump to label 'table_error' [-fpermissive]
 table_error:
 ^~~~~~~~~~~
/home/venki/work/ps/7617/sql/auth/sql_user_table.cc:1614:12: note:   from here
       goto table_error;
            ^~~~~~~~~~~
/home/venki/work/ps/7617/sql/auth/sql_user_table.cc:1646:18: note:   crosses initialization of 'timeval tm'
   struct timeval tm = thd->query_start_timeval_trunc(0);
```
As it is possible to skip the initialization because of a `goto` in the same function.

Reply:
ah, gotos dont like initialization in middle. You can ingore